### PR TITLE
Enables more security headers by default

### PR DESCRIPTION
**Pull request by javitonino**
I took the chance to enable some security headers. More interesting is the `X-Frame-Options` which was not enabled for APIs (it doesn't matter there) and the new `BuilderController` sub-classes.

The other ones should make the loading of `.js` and `.css` safer, but should not have an effect on normal use of the application as it is. We just need to check that nothing broke (best way is to check browser console in each of the main sections of the application).

**Deploy**
Please notify support. They should be made aware that we may have new issues about embedding CARTO pages inside iframes.

**Acceptance**
- [x] Check that you cannot embed (in an iframe) dashboard and friends. Check the browser console for errors.
- [x] Check that you cannot embed (in an iframe) account pages. Check the browser console for errors.
- [x] Check that you cannot embed (in an iframe) Builder. Check the browser console for errors.
- [x] Check that you can embed (in an iframe) editor and Builder embeds. Check the browser console for errors.

### CODE REVIEW

**Comment by juanignaciosl** [comment]
🔴

File: app/controllers/admin/visualizations_controller.rb [hunk 1]
```diff
@@ -715,4 +714,8 @@ def redirect_to_builder_embed_if_v3
       redirect_to CartoDB.url(self, 'builder_visualization_public_embed', visualization_id: @visualization.id)
     end
   end
+
+  def x_frame_options_allow
```
Comment (juanignaciosl) [inline app/controllers/admin/visualizations_controller.rb]: What about moving this to a helper, as it's needed in several controllers?

**Comment by javitonino** [comment]
Extracted to a helper. I think the risk to break things is small, it only affects APIs (which should not be embedded in iframes anyway) and the dataset/Builder views. But I'll add to the ticket header a notice to notify support 👍

**Comment by juanignaciosl** [comment]
👍

**Pull request merged**